src/main/scala/viper/gobra/frontend/InformativeErrorListener.scala

```diff
@@ -0,0 +1,246 @@
+      case null => context.msg match {
+        case extraneous() => analyzeExtraneous(context)
+        case missing() => DefaultErrorType()
+      }
+      case _ => DefaultErrorType()
+    }
+  }
+
+
+  def analyzeFailedPredicate(implicit context: ParserErrorContext, exception: FailedPredicateException): ErrorType = {
+    val parser = context.recognizer
+    parser.getContext match {
+      // One example of a known pattern: Parser reads ':=' when expecting the end of statement: The user probably
+      // used ':=' instead of '='
+      case _ : GobraParser.EosContext => {
+        context.offendingSymbol.getType match {
+          case GobraParser.DECLARE_ASSIGN => GotAssignErrorType()
+          case _ => DefaultFailedEOS()
+        }
+      }
+      case _ => DefaultErrorType()
+    }
+  }
+
+  def analyzeExtraneous(implicit context: ParserErrorContext): ErrorType = {
+    (context.offendingSymbol.getType, context.recognizer.getContext) match {
+      case (GobraParser.DECLARE_ASSIGN, _ : TypeSpecContext) => GotAssignErrorType()
+      case (GobraParser.R_BRACKET, expr : ExpressionContext) if expr.parent.isInstanceOf[CapContext] => SliceMissingIndex(3)
+      case _ => DefaultExtraneous()
+    }
+  }
+
+  def analyzeInputMismatch(implicit context: ParserErrorContext, exception: InputMismatchException): ErrorType = {
+    (context.offendingSymbol.getType, context.recognizer.getContext) match {
+      case (Token.EOF, _) => DefaultMismatch()
+      case (s, i : ImplementationProofContext) if context.recognizer.getExpectedTokens == IntervalSet.of(GobraParser.IMPL)=> {
+        IgnoreError()
+      }
+      case (GobraParser.DECLARE_ASSIGN, _) => GotAssignErrorType()
+      case (GobraParser.R_BRACKET, e : ExpressionContext) if e.parent.isInstanceOf[CapContext] => SliceMissingIndex(3)
+      case _ => DefaultMismatch()
+    }
+  }
+
+  def analyzeNoViable(implicit context: ParserErrorContext, exception: NoViableAltException): ErrorType = {
+    val parser = context.recognizer
+    val ctx = parser.getContext
+    ctx match {
+      case slice : Slice_Context => {
+        // Missing either the second or second and third argument (or completely wrong)
+        SliceMissingIndex()
+      }
+      case _ : VarSpecContext | _ : Type_Context if context.offendingSymbol.getType == GobraParser.DECLARE_ASSIGN => GotAssignErrorType()
+      case eos : EosContext => {
+        parser.getTokenStream.LT(2).getType match {
+          case GobraParser.DECLARE_ASSIGN => GotAssignErrorType()(context.copy(offendingSymbol = parser.getTokenStream.LT(2)))
+          case _ => DefaultNoViable(exception)
+        }
+      }
+      case e : ExpressionContext if e.parent.isInstanceOf[CapContext] => SliceMissingIndex(3)
+      case _ if new_reserved.contains(context.offendingSymbol.getType) => ReservedWord()
+      case _ => DefaultNoViable(exception)
+    }
+  }
+
+
+  protected def underlineError(context : ErrorContext, restOfTheLine : Boolean = false): String = {
```

Comment:
I do not understand what this is doing. If something mostly corresponds to a default ANTLR implementation and you only changed a specific part, then please clarify this. This holds for the entire PR.